wolfcrypt/src/integer.c

```diff
@@ -328,7 +317,7 @@ int mp_copy (mp_int * a, mp_int * b)
   }
 
   /* grow dest */
-  if (b->alloc < a->used) {
+  if (b->alloc < a->used || b->alloc == 0) {
```

Comment:
Required because mp_copy failed to allocate buffer for mp_int for case where used == 0, which later causes issues because the ->dp buffer doesn't exist.